airflow/models/baseoperator.py

```diff
@@ -298,7 +294,7 @@ def __init__(
         end_date: Optional[datetime] = None,
         depends_on_past: bool = False,
         wait_for_downstream: bool = False,
-        dag: Optional[DAG] = None,
+        dag=None,
```

Comment:
I think we decided in one of the PR to keep the explicit `Optional` for default None values too

Reply:
So worth keeping 
```suggestion
        dag: Optional[DAG] = None,
```

Reply:
Imported DAG at the __init__ level.

Reply:
Does this mean that thedag arg doesn't have a type hint anymore? Does `dag: Optional["DAG"]` (or `dag: Optional["airflow.models.DAG"]` help here)?